### ISSUE TITLE
not work in celeba image data in pytorch, but it works well when use torch

### ISSUE THREAD

**Issue by universelee** [junyanz/pytorch-CycleGAN-and-pix2pix#91]
Hi,
in your torch version of CycleGAN, it works well when I want to transfer "not smile" image to "smile" image.
But in pytorch version, it does not work, the result often seems to be a sort of "inverse color" effect. Such effect also occurs when I train on other datasets.
In the page, I can see "This is our ongoing PyTorch implementation", is the pytorch version unfinished, or there is something wrong I missed?

**Comment by junyanz** [comment]
Yeah, it is still an ongoing project. We recently fixed a few issues/differences between PyTorch and Torch, and the results look better. I wonder when you downloaded the code, and if you have tried the latest one.

**Comment by hanrelan** [comment]
I used a recent version of the code (a couple days old) with two face datasets (lfw -> anime) and saw the same issue

**Comment by junyanz** [comment]
@hanrelan Thanks for your nice words. We also observed the color inversion in the PyTorch version, and we are still investigating it. For now, you can add a small [identity](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/master/options/train_options.py#L28) loss to address it.

**Comment by junyanz** [comment]
@hanrelan We also fixed a few initialization issues. And it should work even without identity loss now.

**Comment by mishurov** [comment]
I'm using TFGAN API for TensorFlow
https://ai.googleblog.com/2017/12/tfgan-lightweight-library-for.html

It's quite possible that the issue is implementation specific however the information may be useful.

I train CycleGAN on horse2zebra dataset, I augment it with random left-right flips and random crop after resizing an image to 286x286 and shuffling the dataset.

I've noticed that the "inverse colour" effect appears after 100 steps with batch 1 and it depends on the first images drawn from the dataset. I've solved it by manually tuning random seed for flips, crops and shuffle. If the network doesn't start to invert colours after first 100 steps, it's going to be OK for the rest of training. With image summaries in TensorBoard it's quite easy to tune and restart training couple of times until you get a desirable result.

I also started to train 256x256 images with 6 resnet blocks simply because I couldn't cram a deeper network into 2gb memory of my laptop's GeForce. Apart from horses it started also to recolour green grass into brown soil. The network with 9 resnet blocks on Google Colab does nice though. It could be because of the shallow architecture but it could be because of "unfortunate" combination of images from dataset at the start of training as well.

Thus it could make sense to make random seed a hyperparameter and a cmd line option for datasets.

**Comment by IEWbgfnYDwHRoRRSKtkdyMDUzgdwuBYgDKtDJWd** [comment]
any tips on the best way to go about setting up a colab mishurov? Looked through your repos, figured you would know lol.

**Comment by junyanz** [comment]
Here is the [colab](https://colab.research.google.com/drive/1Enc-pKlP4Q3cimEBfcQv0B_6hUvjVL3o?sandboxMode=true) for cyclegan.

**Comment by mishurov** [comment]
@IEWbgfnYDwHRoRRSKtkdyMDUzgdwuBYgDKtDJWd 

There's no "lol" in my repos. If you find that there, then it's your problem.

**Comment by mishurov** [comment]
I really don't like conversations that way.

I'm straight. I'm not a pervert. Do your perversities to your lazy yanks.

Your code conduct.

I do code.
